datastore/cloud-ndb/named_db.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START datastore_ndb_named_db]
+import os
+
+from google.cloud import ndb
+
+database = os.environ.get("DATASTORE_DATABASE", "prod-database")
+client = ndb.Client(database=database)
+# [END datastore_ndb_named_db]
+
+
+class Book(ndb.Model):
+    title = ndb.StringProperty()
+
+
+def list_books():
+    with client.context():
+        books = Book.query()
+        for book in books:
+            print(book.to_dict())
```

Comment:
It feels a bit unnecessary to have this all this books functionality here when it's not exposed in the sample. And the tests are all surrounding this part of the code, and not testing the functionality covered by the sample itself.

Could we change the sample to return the client, and then write tests against the client directly?

```python
def create_named_db_client():
    # [START datastore_ndb_named_db]
    import os
    from google.cloud import ndb

    database = os.environ.get("DATASTORE_DATABASE", "prod-database")
    client = ndb.Client(database=database)
    # [END datastore_ndb_named_db]
    return client

def test_named_db():
    os.environ["DATASTORE_DATABASE"] = "test"
    client = create_named_db_client()
    assert client.database = "test"
```